modules/dhcp_common/isc/omapi_provider.rb

```diff
@@ -195,22 +195,25 @@ def poap_options_statements(options)
       statements
     end
 
-    def bootServer server
+    def bootServer(server)
       begin
-        ns = ip2hex validate_ip(server)
+        ns = ip2hex(validate_ip(server))
       rescue
         begin
-          ns = ip2hex Resolv.new.getaddress(server)
-        rescue
-          logger.warn "Failed to resolve IP address for #{server}"
+          logger.info "Next-server option not IPv4, trying to resolve PTR record for #{server}"
+          ns = ip2hex(dns_resolv.getaddress(server))
+        rescue Exception => e
```

Comment:
Is there a `ResolvException` or similar we can catch just as a generic resolution failed? That could use `logger.warn` and this could log at the exception level.

Reply:
Yes it's `Resolv::ResolvError`.